packages/venia-ui/lib/components/RichContent/PageBuilder/ContentTypes/ButtonItem/buttonItem.js

```diff
@@ -43,30 +56,83 @@ const ButtonItem = props => {
         paddingLeft
     };
 
-    const cssButtonTypeSuffix =
-        buttonType.charAt(0).toUpperCase() + buttonType.substring(1);
-
     let linkProps = {};
-    let LinkComponent = 'div';
+    let url = '';
     if (typeof link === 'string') {
         linkProps = resolveLinkProps(link, linkType);
-        LinkComponent = linkProps.to ? Link : 'a';
+        url = (linkProps.to ? linkProps.to : linkProps.href).trim();
+    }
+
+    const typeToPriorityMapping = {
+        primary: 'high',
+        secondary: 'normal',
+        link: 'low'
+    };
+
+    const handleClick = useCallback(() => {
+        if (!url) {
+            return;
+        }
+
+        if (openInNewTab) {
+            window.open(url, '_blank');
+        } else if (linkProps.to) {
+            props.history.push(url);
+        } else {
+            window.location.assign(url);
+        }
+    }, [openInNewTab, props.history, url, linkProps.to]);
```

Comment:
Does `history` change often or does it stay the same? If `history` changes often, consider referencing `push` directly instead.

```es6
const { push } = history

const foo = useCallback(() => {
  push()
}, [push])
```